operator/src/main/java/oracle/kubernetes/operator/helpers/KubernetesUtils.java

```diff
@@ -128,14 +130,47 @@ public static boolean isFirstNewer(V1ObjectMeta first, V1ObjectMeta second) {
     DateTime time2 = second.getCreationTimestamp();
 
     if (time1.equals(time2)) {
-      return getResourceVersion(first) > getResourceVersion(second);
+      return getResourceVersion(first).compareTo(getResourceVersion(second)) > 0;
     } else {
       return time1.isAfter(time2);
     }
   }
 
-  private static int getResourceVersion(V1ObjectMeta metadata) {
-    return Integer.parseInt(metadata.getResourceVersion());
+  /**
+   * Parse the resource version from the metadata. According to the Kubernetes design documentation,
+   * https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/
+   *  api-conventions.md#concurrency-control-and-consistency, the resource version is technically opaque; however,
+   * the Kubernetes design also requires that clients be able to list changes to resources "after" the last
+   * change to the same or different resource. Therefore, all Kubernetes implementations use a increasing positive
+   * integer value for the resource version. This can be useful to detect out-of-order watch events. This method
+   * parses the metadata's resource version into a big integer or to 0, if the value is not parsable.
+   * @param metadata Meta data containing resource version
+   * @return The integer value of the resource version or 0, if the value is not parsable
```

Comment:
you say you will return the value or zero, but you seem to return the value or null

Reply:
That's not correct. This method may pass null as the parameter to getResourceVersion(String), which will return 0.

Reply:
ah ok, my bad. thanks